src/main/java/com/labsynch/labseer/domain/StandardizationDryRunCompound.java

```diff
@@ -114,6 +124,99 @@ public StandardizationHistory fetchStats() {
 		return(stats);
 	}
 
+	public static List<Predicate> buildPredicateFromNumericValue(CriteriaBuilder cb, Root<StandardizationDryRunCompound> root, List<Predicate> predicates, String fieldName, Double value, String operator) {
+		Predicate predicate = null;
+		if (operator == null || operator.equals("=")) {
+			predicate = cb.equal(root.get(fieldName), value);
+		} else if (operator.equals(">")) {
+			predicate = cb.greaterThan(root.get(fieldName), value);
+		} else if (operator.equals("<")) {
+			predicate = cb.lessThan(root.get(fieldName), value);
+		}
+		if(predicate != null) {
+			predicates.add(predicate);
+		}
+		return predicates;
+	}
+
+	public static TypedQuery<StandardizationDryRunCompound> searchStandardiationDryRun(StandardizationDryRunSearchDTO dryRunSearch) {
+        EntityManager em = StandardizationDryRunCompound.entityManager();
+        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+        CriteriaQuery<StandardizationDryRunCompound> criteria = criteriaBuilder.createQuery(StandardizationDryRunCompound.class);
+        Root<StandardizationDryRunCompound> root = criteria.from(StandardizationDryRunCompound.class);
+        criteria.select(root);
+
+		// Predicate List
+		List<Predicate> predicates = new ArrayList<Predicate>();
+
+		// New mol weight
+		if (dryRunSearch.getNewMolWeight() != null) {
+			predicates = buildPredicateFromNumericValue(criteriaBuilder, root, predicates, "newMolWeight", dryRunSearch.getNewMolWeight().getValue(), dryRunSearch.getNewMolWeight().getOperator());
+		}
+
+		// Delta mol weight
+		if (dryRunSearch.getDeltaMolWeight() != null) {
+			predicates = buildPredicateFromNumericValue(criteriaBuilder, root, predicates, "deltaMolWeight", dryRunSearch.getDeltaMolWeight().getValue(), dryRunSearch.getDeltaMolWeight().getOperator());
+		}
+
+		// Old mol weight
+		if (dryRunSearch.getOldMolWeight() != null) {
+			predicates = buildPredicateFromNumericValue(criteriaBuilder, root, predicates, "oldMolWeight", dryRunSearch.getOldMolWeight().getValue(), dryRunSearch.getOldMolWeight().getOperator());
+		}
+
+		// Corp name in list
```

Comment:
The one thing I see missing based on the requirements is the ability to _exclude_ corp names. I was picturing a list of corp names, then a boolean that controls whether the list gets _included_ or _excluded_

Reply:
Thanks, just added that Boolean option